File: 250217/숫자가 더 큰 인접한 곳으로 이동/move-to-larger-adjacent-cell.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        // Please write your code here.
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int r = sc.nextInt() -1;
        int c = sc.nextInt() -1;

        int[][] grid = new int[n][n];
        for(int i = 0; i < n; i++){
            for(int  j = 0; j < n; j++){
                grid[i][j] = sc.nextInt();
            }
        }
        while(true){
            System.out.print(grid[r][c]);
            System.out.print(' ');
            if(0 <= r-1 && grid[r][c] < grid[r-1][c]) {
                r--;
            } else if(r+1 < n && grid[r][c] < grid[r+1][c]){
                r++;
            } else if(0 <= c-1 && grid[r][c] < grid[r][c-1]) {
                c--;
            } else if(c+1 < n && grid[r][c] < grid[r][c+1]){
                c++;
            } else break;            
        }
    }
}

File: 260324/숫자가 더 큰 인접한 곳으로 이동/move-to-larger-adjacent-cell.py
n, r, c = map(int, input().split())
a = [[0] * (n + 1) for _ in range(n + 1)]

for i in range(1, n + 1):
    row = list(map(int, input().split()))
    for j in range(1, n + 1):
        a[i][j] = row[j - 1]

# Please write your code here.
import copy
import collections

grid = copy.deepcopy(a)

dr = [-1, 1, 0, 0]
dc = [0, 0, -1, 1]

def in_range(r, c):
    return 0 <= min(r,c) and max(r,c) < n

def move_to_bigger(r, c) -> tuple[bool, int, int]:
    for i in range(4):
        rr = r + dr[i]
        cc = c + dc[i]
        if in_range(rr, cc) and grid[r][c] < grid[rr][cc]:
            return (True, rr, cc)
    
    return (False, 0, 0)

history = []
while True:
    history.append(grid[r][c])
    has_bigger, r, c = move_to_bigger(r, c)
    if not has_bigger:
        break

print(*history)